### PR TITLE
Use Microsoft.SqlServer.DacFx rather than sqlpackage

### LINKED ISSUE #2251: Use Microsoft.SqlServer.DACFx nuget instead of sqlpackage.exe
### Description

This is a suggestion to use the new [Microsoft.SqlServer.DACFx](https://www.nuget.org/packages/Microsoft.SqlServer.DacFx/) NuGet package to drive Fake.Sql.DacPac rather than relying on `sqlpackage.exe`. Would this be of interest? I am not familiar with these libraries but am interested in looking into this if it can alleviate some xplat pain with DacPac.

Note also that Microsoft is now providing previews of [`sqlpackage`](https://docs.microsoft.com/en-us/sql/tools/sqlpackage-download?view=sql-server-2017) that can run on linux and macOS.

cc @isaacabraham

### PR DESCRIPTION
**Pull request by panesofglass**
### Description

Changes `Fake.Sql.DacPac` to use the `Microsoft.SqlServer.DacFx` package rather than shelling out to `sqlpackage.exe`.

- fixes #2251 

## TODO

Feel free to open the PR and ask for help

- [ ] New (API-)documentation for new features exist (Note: API-docs are enough, additional docs are in `help/markdown`)
- [ ] unit or integration test exists (or short reasoning why it doesn't make sense)
- [ ] boy scout rule: "leave the code behind in a better state than you found it" (fix warnings, obsolete members or code-style in the places you worked in)

### CODE REVIEW

**Comment by panesofglass** [comment]
I’m not sure how to add meaningful tests for this. Thoughts?

File: src/app/Fake.Sql.DacPac/Sql.DacPac.fs [hunk 1]
```diff
@@ -21,8 +21,6 @@ module DacPac =
 
     /// Configuration arguments for DacPac deploy
     type DeployDbArgs = {
-        /// The path to SqlPackage.exe.
-        SqlPackageToolPath : string
```
Comment (matthid) [inline src/app/Fake.Sql.DacPac/Sql.DacPac.fs]: This is basically a breaking change, should we do that, or just leave the field and add a warning if it is set?

File: src/app/Fake.Sql.DacPac/Sql.DacPac.fs [hunk 2]
```diff
@@ -42,25 +40,9 @@ module DacPac =
...
-        let vs = !!(Environment.ProgramFilesX86 </> @"Microsoft Visual Studio*\Common7\IDE\Extensions\Microsoft\SQLDB\DAC\*\SqlPackage.exe") |> Seq.map(fun path -> path, getVsVersion path)
-        let vs2017 = !!(Environment.ProgramFilesX86 </> @"Microsoft Visual Studio\**\Common7\IDE\Extensions\Microsoft\SQLDB\DAC\*\SqlPackage.exe") |> Seq.map(fun path -> path, getVsVersion path)
-
-        [ sql; vs; vs2017 ]
-        |> List.collect Seq.toList
-        |> List.sortByDescending snd
-        |> List.map fst
-
     /// The default DacPac deployment arguments.
     let DefaultDeploymentArgs =
```
Comment (matthid) [inline src/app/Fake.Sql.DacPac/Sql.DacPac.fs]: If we already decide to be breaking, we can change this to `internal`

File: src/app/Fake.Sql.DacPac/Sql.DacPac.fs (hunk 1 above)
Comment (panesofglass) [reply src/app/Fake.Sql.DacPac/Sql.DacPac.fs]: The latter is probably the better idea. I’ll add back this field.

**Comment by panesofglass** [comment]
Having replaced the Fake.Sql.DacPac with Microsoft.SqlServer.DacFx directly, I don't think it meaningful to replace it here. The outputs are slightly different, and it would be best to modify what's here more significantly. I think it's best to leave this as-is and let those who want the Microsoft.SqlServer.DacFx approach to just use it directly. It's roughly equivalent code with stronger type safety.

**Comment by matthid** [comment]
Question is: should we mark the module obsolete and point people to the alternative, maybe even remove it from the code base?

**Pull request closed without merging**